components/bsx/Offer/MasterOfferTable.vue

```diff
@@ -55,161 +54,130 @@
-import { Offer, OfferResponse } from './types'
-import PrefixMixin from '~/utils/mixins/prefixMixin'
-import SubscribeMixin from '~/utils/mixins/subscribeMixin'
+<script lang="ts" setup>
+import { OfferResponse } from './types'
 import offerListUser from '@/queries/subsquid/bsx/offerListUser.graphql'
 import { notificationTypes, showNotification } from '~/utils/notification'
 import { isAddress } from '@polkadot/util-crypto'
-import ChainMixin from '~/utils/mixins/chainMixin'
 import { SelectedOfferType } from '~/utils/offerStatus'
 
-const components = {
-  Loader: () => import('@/components/shared/Loader.vue'),
-  CollapseCardWrapper: () =>
-    import('@/components/shared/collapse/CollapseCardWrapper.vue'),
-  StatsOverview: () => import('~/components/bsx/Offer/StatsOverview.vue'),
-  AddressInput: () => import('@/components/shared/AddressInput.vue'),
-  OffersUserTable: () => import('@/components/bsx/Offer/OffersUserTable.vue'),
-  MyOffer: () => import('@/components/bsx/Offer/MyOffer.vue'),
-  OfferTable: () => import('@/components/bsx/Offer/OfferTable.vue'),
-}
-
-@Component({ components })
-export default class MasterOfferTable extends mixins(
-  AuthMixin,
-  MetaTransactionMixin,
-  PrefixMixin,
-  SubscribeMixin,
-  ChainMixin
-) {
-  protected createdOffers: Offer[] = []
-  protected total = 0
-  protected destinationAddress = ''
-  protected accountIdChanged = ''
-  protected incomingOffers: Offer[] = []
-  protected SelectedOfferType = SelectedOfferType
-  protected skipUserOffer = false // skip fetching with id when this variable is true
-
-  public async created() {
-    this.checkQueryParams()
-  }
-
-  fetch() {
-    this.fetchCreatedOffers()
-  }
-
-  get isOfferDropdownDisabled(): boolean {
-    return this.skipUserOffer
-  }
-
-  get ss58Format(): number {
-    return this.chainProperties?.ss58Format
-  }
-
-  get selectedOfferType(): string {
-    return (this.$route.query.tab as string) || SelectedOfferType.ALL
-  }
-
-  set selectedOfferType(val) {
-    const { target } = this.$route.query
-    this.$router.replace({
-      query: { target, tab: val },
+import OfferTable from '@/components/bsx/Offer/OfferTable.vue'
+import OffersUserTable from '@/components/bsx/Offer/OffersUserTable.vue'
+import MyOffer from '@/components/bsx/Offer/MyOffer.vue'
+
+const createdOffers = ref([])
+const destinationAddress = ref('')
+const accountIdChanged = ref('')
+const incomingOffers = ref([])
+const skipUserOffer = ref(false) // skip fetching with id when this variable is true
+
+const { $apollo, $route, $router } = useNuxtApp()
+const { accountId, isLogIn } = useAuth()
+const { client } = usePrefix()
+
+const selectedOfferType = computed({
+  get: () => $route.query.tab || SelectedOfferType.ALL,
+  set: (value) => {
+    const { target } = $route.query
+    $router.replace({
+      query: { target, tab: value },
     })
+  },
+})
+
+const isOfferDropdownDisabled = computed(() => skipUserOffer.value)
+
+const OfferTypeOptions = ref([
```

Comment:
sir we know each other for a long time, 
but having variable with big letter makes me sad

Reply:
Pascal Case ftw